schemas/create-task-request.yml

```diff
@@ -13,6 +13,15 @@ properties:
     minLength:      {$const: identifier-min-length}
     maxLength:      {$const: identifier-max-length}
     pattern:        {$const: identifier-pattern}
+  clientId:
```

Comment:
@djmitche, why is this part of the task definition?

Could we maybe make it part of `task.metadata` and then just infer it from the caller?

I know this is a design choice, so I'm curious what the pros/cons are...

---
Other question, when issuing temporary credentials to workers in `claimWork` should we re-use the task `clientId`?

That way if I use `my-client-id` to create taskA and taskA uses `task.scopes` to do something bad the audit log will say `my-client-id` did it.

Sorry, I'm bring up the design here, but implementation is probably a detail, agreeing what we should do here is the harder part.

Reply:
Putting it in metadata's not a bad idea (and not a large change to the implementation).

I'd be OK using this as *part* of the clientId in the claim credentials, but those already include a taskId so it would really only save a step -- and in situations where task A creates task B creates task C .. it could potentially produce very long clientIds.  Probably best to avoid that.